paper/VIT/train_memory_vit.py

```diff
@@ -0,0 +1,245 @@
+parser.add_argument("--patch", default=8, type=int)
+parser.add_argument("--batch-size", default=128, type=int)
+parser.add_argument("--eval-batch-size", default=1024, type=int)
+parser.add_argument("--lr", default=1e-3, type=float)
+parser.add_argument("--min-lr", default=1e-5, type=float)
+parser.add_argument("--beta1", default=0.9, type=float)
+parser.add_argument("--beta2", default=0.999, type=float)
+parser.add_argument("--off-benchmark", action="store_true")
+parser.add_argument("--max-epochs", default=200, type=int)
+parser.add_argument("--dry-run", action="store_true")
+parser.add_argument("--weight-decay", default=5e-5, type=float)
+parser.add_argument("--warmup-epoch", default=5, type=int)
+parser.add_argument("--precision", default=16, type=int)
+parser.add_argument("--autoaugment", action="store_true")
+parser.add_argument("--criterion", default="ce")
+parser.add_argument("--label-smoothing", action="store_true")
+parser.add_argument("--smoothing", default=0.1, type=float)
+parser.add_argument("--rcpaste", action="store_true")
+parser.add_argument("--cutmix", action="store_true")
+parser.add_argument("--mixup", action="store_true")
+parser.add_argument("--dropout", default=0.0, type=float)
+parser.add_argument("--head", default=12, type=int)
+parser.add_argument("--num-layers", default=7, type=int)
+parser.add_argument("--hidden", default=384, type=int)
+parser.add_argument("--mlp-hidden", default=384, type=int)
+parser.add_argument("--off-cls-token", action="store_true")
+parser.add_argument("--seed", default=42, type=int)
+parser.add_argument("--mem_samples", default=100, type=int)
+parser.add_argument("--data_samples", default=1000, type=int)
+parser.add_argument("--project-name", default="VisionTransformer")
+parser.add_argument("--data_root", default="data", type=str)
+args = parser.parse_args()
+torch.manual_seed(args.seed)
+np.random.seed(args.seed)
+args.benchmark = True 
+args.gpus = torch.cuda.device_count()
+args.num_workers = 4*args.gpus if args.gpus else 8
+args.is_cls_token = True if not args.off_cls_token else False
+
+
+def set_seed(seed):
+    """ Method to set seed across runs to ensure reproducibility.
+    It fixes seed for single-gpu machines.
+
+    Args:
+        seed (int): Seed to fix reproducibility. It should different for
+            each run 
+
+    Returns:
+        RandomState: fixed random state to initialize dataset iterators
+    """
+    torch.backends.cudnn.deterministic = True
+    torch.backends.cudnn.benchmark = False # set to false for reproducibility, True to boost performance
+    torch.manual_seed(seed)
+    np.random.seed(seed)
+    torch.cuda.manual_seed(seed)
+    random.seed(seed)
+    random_state = random.getstate()
+    os.environ["CUBLAS_WORKSPACE_CONFIG"] = ":4096:8"
+    return random_state
+def split_dataset(dataset:torch.utils.data.Dataset,train_size:int,val_size:int,seed:int)->List[torch.utils.data.Dataset]:
```

Comment:
please space the methods. Could you parse the whole repo with the black formatter?

Reply:
Addressed in [0c5f7a0](https://github.com/KRLGroup/memory-wrap/pull/3/commits/0c5f7a06097775712e22d547f8b01386f699e196)